packages/angular-query-experimental/src/providers.ts

```diff
@@ -0,0 +1,86 @@
+import { InjectionToken, makeEnvironmentProviders } from '@angular/core'
+import { createQuery } from './createQuery'
+import { QUERY_CLIENT } from './queryClient'
+import { QueryClientService } from './QueryClientService'
+import { createMutation } from './createMutation'
+import { createQueries } from './createQueries'
+import { useIsFetching } from './useIsFetching'
+import { useIsMutating } from './useIsMutating'
+import type { EnvironmentProviders } from '@angular/core'
+import type { QueryClient } from '@tanstack/query-core'
+
+export const CreateQuery = new InjectionToken<typeof createQuery>('CreateQuery')
+export const CreateMutation = new InjectionToken<typeof createMutation>(
+  'CreateMutation',
+)
+export const CreateQueries = new InjectionToken<typeof createQueries>(
+  'CreateQueries',
+)
+export const UseQueryClient = new InjectionToken<QueryClient>('UseQueryClient')
+export const UseIsFetching = new InjectionToken<typeof useIsFetching>(
+  'UseIsFetching',
+)
+export const UseIsMutating = new InjectionToken<typeof useIsMutating>(
+  'UseIsMutating',
+)
+
+/**
+ * Sets up providers necessary to enable Angular Query functionality for the application.
+ * Allows to configure the query client.
+ *
+ * @usageNotes
+ *
+ * Basic example of how you can provide the query client to your application:
+ * ```
+ * bootstrapApplication(AppComponent, {
+ *   providers: [provideAngularQuery(new QueryClient())]
+ * });
+ * ```
+ *
+ * You can also configure the query client by passing the optional configuration object to its constructor:
+ * ```
+ * bootstrapApplication(AppComponent, {
+ *   providers: [
+ *     provideAngularQuery(
+ *       new QueryClient({
+ *         // config options
+ *       }),
+ *     ),
+ *   ],
+ * })
+ * ```
+ *
+ *
+ * @publicApi
+ * @param queryClient The `QueryClient` to provide.
+ * @returns A set of providers to set up Angular Query.
+ */
+export function provideAngularQuery(
+  queryClient: QueryClient,
```

Comment:
I'd like the configuration to be an object, so later we can pass other configs. 

```ts
{
  client: QueryClient,
  // defaultOptions ?
}
```

Reply:
I started with implementing this but I think I will pass on this one: looking at for example `provideRouter` in Angular itself it works with separate parameters instead of an options object.

`defaultOptions` needs to be passed to the `QueryClient` constructor (which comes straight from Tanstack query-core) and I do not foresee any other configuration will be needed. It would also mean an extra nesting level. Using an overload your suggestion could be implemented in a non-breaking way later.